src/flb_io.c

```diff
@@ -63,6 +63,7 @@
 #include <fluent-bit/flb_network.h>
 #include <fluent-bit/flb_engine.h>
 #include <fluent-bit/flb_thread.h>
+#include <fluent-bit/flb_http_client.h>
```

Comment:
the ```io``` interface should not be aware about HTTP contexts. Can you move the new functions to the flb_http_client.c ?

note: make sure to prefix the functions with the proper interface, e.g: ```flb_http_client_proxy_connect```